### DISCUSSION TITLE
Question regarding `es` and coercion

### DISCUSSION THREAD

**Discussion by davidgarland** [haskell-effectful/effectful#124] [Q&A] [answered]
Is it safe to use `unsafeCoerce` to turn `Eff '[] a` into `Eff es a`? I would expect that the two have the same runtime representation, but I'm just making sure.

I know in general I should just avoid unifying `es` with `'[]` for as long as possible, but due to a design problem out of my control I cannot avoid this. (In specific, I'm trying to write a sort of iterator library, and implementing `IsList` for those iterators requires that there are no effects present, because `IsList` has `toList` as a member rather than that being its own class, making the type of `fromList` unnecessarily restrictive-- being allowed to cast `Eff '[] a` to `Eff es a` would at least allow users to get effectual code that uses list literals to work)

**Comment by arybczak** [comment, marked as answer]
You don't have to use `unsafeCoerce`, you can use `inject`, but you need a version from master, i.e. yet unreleased 2.2.2.0.

I'll release it in a couple of days.

**Reply by davidgarland** [reply]
Thank you for the help, it is much appreciated :)
